noir-projects/noir-contracts/contracts/stateful_test_contract/src/main.nr

```diff
@@ -46,7 +46,8 @@ contract StatefulTest {
     fn create_note_no_init_check(owner: AztecAddress, value: Field) {
         if (value != 0) {
             let loc = storage.notes.at(owner);
-            increment(loc, value, owner);
+            // TODO (#6312): This will break with key rotation. Fix this. Will not be able to spend / increment any notes after rotating key.
```

Comment:
Why would it be unable to increment after key rotation here? If it is using the utils it would be using the npk_m_hash, but it is related to the decrease then as the utils where broken, right? Might be fine to not have here as it don't need to be handled here but in the utils where there already was a todo.